### PR TITLE
refactor: Signet fixups

### PR DESCRIPTION
**Pull request by maflcko**
Some doc and test fixups for #18267

### CODE REVIEW

File: doc/bips.md [hunk 1]
```diff
@@ -42,4 +42,5 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v0.21.0**):
 * [`BIP 173`](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki): Bech32 addresses for native Segregated Witness outputs are supported as of **v0.16.0** ([PR 11167](https://github.com/bitcoin/bitcoin/pull/11167)). Bech32 addresses are generated by default as of **v0.20.0** ([PR 16884](https://github.com/bitcoin/bitcoin/pull/16884)).
 * [`BIP 174`](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki): RPCs to operate on Partially Signed Bitcoin Transactions (PSBT) are present as of **v0.17.0** ([PR 13557](https://github.com/bitcoin/bitcoin/pull/13557)).
 * [`BIP 176`](https://github.com/bitcoin/bips/blob/master/bip-0176.mediawiki): Bits Denomination [QT only] is supported as of **v0.16.0** ([PR 12035](https://github.com/bitcoin/bitcoin/pull/12035)).
+* [`BIP 325`](https://github.com/bitcoin/bips/blob/master/bip-0325.mediawiki): Signet test network is supported as of **v0.21.0** ([PR 18267](https://github.com/bitcoin/bitcoin/pull/18267)).
```
Comment (ajtowns) [inline doc/bips.md:45]: (Not strictly true until bitcoin/bips#978 and bitcoin/bips#1000 are merged)

File: doc/release-notes.md [hunk 2]
```diff
@@ -335,6 +335,10 @@ RPC
 Tests
 -----
 
+- The BIP 325 default signet can be enabled by the `-chain=signet` or `-signet`
+  setting. The settings `-signetchallenge` and `-signetseednode` allow
+  modifying the network.
```
Comment (ajtowns) [inline doc/release-notes.md:340]: "modifying the network" -> "enabling a custom signet instead" ?

File: src/signet.cpp [hunk 3]
```diff
@@ -65,8 +65,9 @@ static uint256 ComputeModifiedMerkleRoot(const CMutableTransaction& cb, const CB
     return ComputeMerkleRoot(std::move(leaves));
 }
 
-SignetTxs SignetTxs::Create(const CBlock& block, const CScript& challenge)
+Optional<SignetTxs> SignetTxs::Create(const CBlock& block, const CScript& challenge)
 {
+    const auto invalid = []() -> Optional<SignetTxs> { return nullopt; };
```
Comment (ajtowns) [inline src/signet.cpp:70]: Should just replace `return invalid()` with `return nullopt` directly.

File: src/signet.h [hunk 4]
```diff
@@ -22,21 +24,14 @@ bool CheckSignetBlockSolution(const CBlock& block, const Consensus::Params& cons
...
-
     template<class T1, class T2>
-    SignetTxs(const T1& to_spend, const T2& to_sign) : m_to_spend{to_spend}, m_to_sign{to_sign}, m_valid(true) { }
-
-    static SignetTxs Create(const CBlock& block, const CScript& challenge);
+    SignetTxs(const T1& to_spend, const T2& to_sign) : m_to_spend{to_spend}, m_to_sign{to_sign} { }
 
 public:
-    SignetTxs(const CBlock& block, const CScript& challenge) : SignetTxs(Create(block, challenge)) { }
+    static Optional<SignetTxs> Create(const CBlock& block, const CScript& challenge);
```
Comment (ajtowns) [inline src/signet.h:31]: This is much better!

**Review by ajtowns — COMMENTED** [review]
Concept ACK

File: doc/bips.md (hunk 1 above)
Comment (MarcoFalke) [reply doc/bips.md:45]: heh, I hope that those are merged before we release the next major version ;)

Comment (kallewoof) [reply doc/bips.md:45]: I've ACK'd both, so they should be considered merged, I think, since i'm the BIP author.

**Comment by fjahr** [comment]
@ajtowns suggestions are good, ACK otherwise.

File: src/signet.cpp (hunk 3 above)
Comment (MarcoFalke) [reply src/signet.cpp]: thx done

File: doc/release-notes.md (hunk 2 above)
Comment (MarcoFalke) [reply doc/release-notes.md]: thx done

**Comment by fjahr** [comment]
Code review ACK fadee171c8b0b591daf8f1d65640e4f288b3c3bf

**Review by dr-orlovsky — APPROVED** [review]
Reviewed & ACK https://github.com/bitcoin/bitcoin/pull/19993/commits/facaf9e61f4b9ea91fab554d495ebea1043d08fb

**Comment by ajtowns** [comment]
ACK facaf9e61f4b9ea91fab554d495ebea1043d08fb -- code review only

**Review by kallewoof — COMMENTED** [review]
Code review ACK facaf9e61f4b9ea91fab554d495ebea1043d08fb

**Pull request merged**